### PR TITLE
Updates for GraphQL api spec

### PR DESCRIPTION
**Pull request by cafca**
- Make document metadata field generic over schemas: We don't have anything schema-specific in this type so it can be generic.
- Use GraphQL language (e.g. "field argument" instead of "input variable") and camel case
- Specify types of field values
- Simplify GraphQL type and field names

### CODE REVIEW

**Comment by adzialocha** [comment]
How about calling it just `DocumentMeta`?

**Comment by cafca** [comment]
That's what I called it before and then I made it `DocumentMetadata` because I thought you will probably not like the abbreviation :D

**Comment by cafca** [comment]
You should own that term! I love metadata, it's the best :) Don't leave it to big tech.

File: website/docs/specification/networking/queries.md [hunk 1]
```diff
@@ -111,53 +111,50 @@ type NextEntryArguments {
 
 ## Querying documents
 
-- these queries allow clients to request the field contents of materialised document views and metadata for their associated documents
-- some GraphQL operations and types are dynamic in that they depend on the schemas known to the node
-  - this spec only gives a generic form for these operations and types
-  - in this specification we use `<schema_id>` as a placeholder for the string-encoded schema id of actual schemas
+- For every schema that can be queried, nodes generate fields, which are made available on the _root query type_, as well as types to represent the schema's documents.
+  - Together, these allow clients to request documents including their materialised views and metadata.
+- As fields and types that contain schema-specific data are dynamically generated, every node may contain a different set of these, depending on which schemas are available.
```
Comment (adzialocha) [inline website/docs/specification/networking/queries.md:116]: Not sure if it's clear _what_ varies here: is it the response (node a has 15 documents, b has only 3), the offered queries (node a offers schema p, node b offers schema q and p) or features of them (node a offers meta information about authors, b doesn't)?

File: website/docs/specification/networking/queries.md [hunk 2]
```diff
@@ -174,24 +171,41 @@ type <schema_id>ResponseMeta {
...
-  ... potentially more fields
-  """
+- returns a single document that uses this schema
+  - implementations must have no side effects
+- the name of this field is equal to the _schema id_ of the schema it represents
+- either the `id` or `viewId` field argument must be set
+  - if `id` contains a document id, the response must contain the [_latest document view_][latest-document-view] for that document
+  - if `viewId` contains a document view id, the response must contain this document view
+  - if both field arguments are given the query must return an error
+- not every node holds all documents and especially not all document views (historical states of a document) in its database because of the decentralised nature of p2panda. in this case a "not found" error will be returned
```
Comment (adzialocha) [inline website/docs/specification/networking/queries.md:185]: Probably a good idea to also specify how errors look like and what they contain?  Maybe not for this PR but I can make an issue if you agree.

File: website/docs/specification/networking/queries.md (hunk 1 above)
Comment (cafca) [reply website/docs/specification/networking/queries.md:116]: I think this is pretty unambiguous already. It says that _the set_ of schema-specific fields and types varies.

File: website/docs/specification/networking/queries.md (hunk 2 above)
Comment (cafca) [reply website/docs/specification/networking/queries.md:185]: Yes, definitely!

File: website/docs/specification/networking/queries.md (hunk 1 above)
Comment (adzialocha) [reply website/docs/specification/networking/queries.md:116]: > It says that the set of schema-specific fields and types varies.

I think I'm still lost here, haha. What are "types"?

File: website/docs/specification/networking/queries.md (hunk 2 above)
Comment (cafca) [reply website/docs/specification/networking/queries.md:185]: https://github.com/p2panda/handbook/issues/206

File: website/docs/specification/networking/queries.md (hunk 1 above)
Comment (cafca) [reply website/docs/specification/networking/queries.md:116]: Hmm, maybe the confusion comes from GraphQL terminology? The structure of a GraphQL schema is that at the root you have the _root query type_ which is an _object_. Like every object in GraphQL it has _fields_ that can be queried to access: either another object with more fields or a scalar type (that actually has a concrete value). 

These fields are what you might think of as "queries" in another context, e.g. `nextArgs` in GraphQL is not a query, it's a _field_ on the query object.

So all the fields on the root query type change depending on what schemas are available. But also in other places: for example the available fields on `inbound` also change depending on the available schemas. That's why I wrote it like this. I will try and rephrase the sentence to try and make this clearer, please let me know if it works :D

File: website/docs/specification/networking/queries.md [hunk 3]
```diff
@@ -111,53 +111,53 @@ type NextEntryArguments {
...
-    - implementations must have no side effects
-- either the `id` or `view_id` input variable must to be set
-  - if `id` contains a document id the response must contain the [_latest document view_][latest-document-view] for that document
-  - if `view_id` contains a document view id, the query must contain this document view
-  - if both input variables are given the query must return an error
-- not every node holds all documents and especially not all document views (historical states of a document) in its database because of the decentralised nature of p2panda. in this case a "not found" error will be returned
+- Nodes generate two GraphQL types for every schema that can be queried:
+  1. a type `<schema_id>` that contains fields for document metadata and the associated document view
+  2. a type `<schema_id>Fields` to represent the document view's fields with the actual data contained in the document
+- Document fields with the types `Boolean`, `Integer`, `Float` and `Text` are represented with the corresponding GraphQL scalar types.
```
Comment (sandreae) [inline website/docs/specification/networking/queries.md:128]: `String`

File: website/docs/specification/networking/queries.md [hunk 4]
```diff
@@ -174,24 +174,41 @@ type <schema_id>ResponseMeta {
...
-  """
-  ... potentially more fields
-  """
+- returns a single document that uses this schema
+  - implementations must have no side effects
+- the name of this field is equal to the _schema id_ of the schema it represents
+- either the `id` or `viewId` field argument must be set
+  - if `id` contains a document id, the response must contain the [_latest document view_][latest-document-view] for that document
+  - if `viewId` contains a document view id, the response must contain this document view
+  - if both field arguments are given the query must return an error
```
Comment (sandreae) [inline website/docs/specification/networking/queries.md:187]: Change this to current behaviour (return the view)

File: website/docs/specification/networking/queries.md [hunk 5]
```diff
@@ -218,32 +235,36 @@ type <schema_id>ResponseFields {
...
-  cursor identifier for pagination
-  """
-  after: String
-): <schema_id>PageResponse!
+type QueryRoot {
+  # ... other query root fields here ...
+
+  all_<schema_id>(
+    """
+    filter collection of documents
```
Comment (sandreae) [inline website/docs/specification/networking/queries.md:243]: These filters are not implemented (right? :rofl: ) should we still include them?

File: website/docs/specification/networking/queries.md [hunk 6]
```diff
@@ -294,7 +315,7 @@ type <schema_id>Filter {
   <field_name>_lte: <value>
 }
 
-type <schema_id>PageResponse {
+type <schema_id>Page {
```
Comment (sandreae) [inline website/docs/specification/networking/queries.md:318]: Same, should we include unimplemented stuff?

**Pull request merged**